sqldb/sqlc/queries/amp_invoices.sql

```diff
@@ -61,7 +61,7 @@ WHERE (
 
 -- name: UpdateAMPSubInvoiceHTLCPreimage :execresult
 UPDATE amp_sub_invoice_htlcs AS a
-SET preimage = $4
+SET preimage = $5
 WHERE a.invoice_id = $1 AND a.set_id = $2 AND a.htlc_id = (
-    SELECT id FROM invoice_htlcs AS i WHERE i.htlc_id = $3
+    SELECT id FROM invoice_htlcs AS i WHERE i.chan_id = $3 AND i.htlc_id = $4
```

Comment:
Is there an easy extension to the existing `invoices` unit tests that can catch that? So something like inserting HTLCS to two distinct channels, updating the preimage, then ensuring that only one of them is actually changed.

Reply:
So see #9022 this is already covered in itests: ```make itest backend=btcd dbbackend=postgres nativesql=true icase=sendpayment_amp_invoice```. We can add additional unit test coverage too though if you feel this is necessary.